src/notifications/v2/notifications.savedSearch.service.ts

```diff
@@ -0,0 +1,189 @@
+                this.SAVED_SEARCH_NOTIFICATION_EMAIL_TEMPLATE_ID,
+                personalisation,
+                `${
+                    this.SAVED_SEARCH_NOTIFICATION_EMAIL_TEMPLATE_ID
+                }-${new Date().toISOString()}`,
+            );
+
+            notification.emailSent = true;
+            await this.savedSearchNotificationService.updateSavedSearchNotification(
+                notification,
+            );
+        }
+    }
+
+    private createSavedSearchNotifications = async (
+        savedSearches: SavedSearch[],
+        yesterday: DateTime,
+    ) => {
+        let numberOfSearchesWithMatches = 0;
+        for (const savedSearch of savedSearches) {
+            const filterArray: FilterArray = buildSearchFilterArray(
+                savedSearch,
+                yesterday.toJSDate(),
+            );
+
+            if (savedSearch.search_term) {
+                const searchTerm = addSearchTerm(savedSearch.search_term);
+                filterArray.push(searchTerm);
+            }
+
+            const matches =
+                await this.grantService.findGrantsMatchingFilterCriteria(
+                    filterArray,
+                );
+
+            if (matches?.length > 0) {
+                numberOfSearchesWithMatches += 1;
+                await this.savedSearchNotificationService.createSavedSearchNotification(
+                    savedSearch,
+                );
+            }
+            return numberOfSearchesWithMatches;
+        }
+    };
+
+    processSavedSearchMatches = async () => {
+        console.log('Running process new saved search matches...');
+
+        const startTime = performance.now();
+
+        const yesterday = DateTime.now().minus({ days: 1 }).startOf('day');
+        const newGrants = await this.grantService.findGrantsPublishedAfterDate(
+            yesterday.toJSDate(),
+        );
+
+        console.log(
+            `Number of grants added since ${yesterday.toJSDate()}: ${
+                newGrants ? newGrants.length : 0
+            }`,
+        );
+
+        if (newGrants.length > 0) {
+            const savedSearches =
+                await this.savedSearchService.findAllConfirmedSearchesWhereDateRangeIsNullOrOverlaps(
+                    yesterday.toJSDate(),
+                );
+
+            const numberOfSearchesWithMatches =
+                await this.createSavedSearchNotifications(
+                    savedSearches,
+                    yesterday,
+                );
+
+            const endTime = performance.now();
+
+            console.log(
+                `Number of saved saved searches to process: ${
+                    savedSearches ? savedSearches.length : 0
+                }`,
+            );
+            console.log(
+                `Number of saved search notifications created: ${numberOfSearchesWithMatches}`,
+            );
+            console.log(
+                `Task took ${endTime - startTime} milliseconds to run \r\n`,
+            );
```

Comment:
Would ideally reference the scheduled job in each log, easy for them to get lost amongst other things happening concurrently in prod